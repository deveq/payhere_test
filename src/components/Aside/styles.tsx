import styled from 'styled-components';
import {flexCenter} from 'styles/helper';

export const Aside = styled.aside`
    width: 200px;
    background: ${({theme}) => theme.bg_page1};
    padding: 30px;
    box-sizing: border-box;
    flex-shrink: 0;
`;

export const List = styled.ul`
    display: grid;
    gap: 5px;
`;

export const Title = styled.h2`
    font-size: 24px;
    font-weight: bold;
    border-bottom: ${({theme}) => `1px solid ${theme.bg_element4}`};
    color: ${({theme}) => theme.text1};
    padding-bottom: 10px;
    margin-bottom: 20px;
    text-indent: 10px;
`;

export const ListItem = styled.li<{selected: boolean}>`
    background: ${({selected,theme}) => selected ? theme.bg_element1 : 'transparent'};
    padding: 15px 20px;
    border-radius: 10px;
    font-weight: bold;
    color: ${({selected, theme}) => selected ? theme.text2 : theme.text3};
    &:hover,
    &:active {
        background: ${({theme}) => theme.bg_element1};
    };
    cursor: pointer;
`;

export const Subtitle = styled.div`

`;