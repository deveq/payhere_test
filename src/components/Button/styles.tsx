import styled from "styled-components";

type ButtonSize = 'sm' | 'md' | 'lg';
export interface ButtonProps {
    size?: ButtonSize;
    backgroundColor?: string;
    borderColor?: string;
    color?: string;
    showShadow?: boolean;
}

const handleSize = (size?: ButtonSize) => {
    if (size === 'sm') return '0.5rem 1rem';
    if (size === 'md') return '0.75rem 1.5rem';
    if (size === 'lg') return '1rem 2rem';
    return '0.75rem 1.5rem';
}

export const Button = styled.button<ButtonProps>`
    background: ${({backgroundColor, theme}) => 
        backgroundColor || theme.bg_element2
    };
    border-radius: 5px;
    color: ${({color, theme}) => 
        color || theme.text1
    };
    padding: ${({size}) => handleSize(size)};
    font-weight: bold;
    box-shadow: ${({theme, showShadow}) => showShadow ? theme.boxShadow : 'none'};
    border: ${({showShadow, theme}) => showShadow ? 'none' : `1px solid ${theme.border2}` };
    &:hover,
    &:active {
        background: ${({theme}) => theme.bg_element3};
    }
    cursor: pointer;
`;

Button.defaultProps = {
    size: 'md',
}

export const Primary = styled(Button)`
    background: ${({theme}) => theme.primary};
    color: white;
`;

export const Destructive = styled(Button)`
    background: ${({theme}) => theme.destructive};
    color: white;
    border: none;
    &:hover,
    &:active {
        background: ${({theme}) => theme.destructive};
    }
`;