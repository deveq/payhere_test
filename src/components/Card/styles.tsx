import styled from 'styled-components';

export const CardContainer = styled.div`
    background: ${({theme}) => theme.bg_element1};
    border-radius: 10px;
    padding: 5px 10px;
`;