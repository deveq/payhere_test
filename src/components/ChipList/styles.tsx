import styled from 'styled-components';

interface ChipProps {
    selected?: boolean;
}

export const Chip = styled.div<ChipProps>`
    padding: 5px 10px;
    max-width: 150px;
    min-width: 40px;
    text-align: center;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
    border-radius: 20px;
    border: 1px solid ${({theme}) => theme.border2};
    background: ${({selected, theme}) => selected ? theme.primary : 'white'};
    color: ${({selected}) => selected ? 'white' : '#333'};
    cursor: pointer;
    &:active {
        opacity: 0.9;
    }
`;

export const ChipList = styled.div`
    display: flex;
    /* flex-direction: column;
    align-items: flex-start; */
    flex-wrap: wrap;
    gap: 8px;
    margin-bottom: 10px;
`;

export const Message = styled.p`
    color: ${({theme}) => theme.text1};
`;