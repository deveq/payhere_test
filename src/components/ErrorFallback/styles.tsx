import styled from 'styled-components';
import {flexCenter} from 'styles/helper';
import {FaRegSadTear} from 'react-icons/fa';

export const Container = styled.div`
    height: 100vh;
    ${flexCenter};
    flex-direction: column;
`;

export const Paragraph = styled.p`
    font-size: 30px;
`;

export const ErrorMessage = styled.pre`
    font-size: 20px;
`;

export const SadIcon = styled(FaRegSadTear)`
    width: 100px;
    height: 100px;
    color: gray;
`;