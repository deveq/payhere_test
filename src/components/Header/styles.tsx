import React from 'react';
import styled from 'styled-components';
import { ReactComponent } from 'assets/logo.svg';
import {flexCenter} from 'styles/helper';


export const Container = styled.header`
	display: flex;
	padding: 1rem 2rem;
	background: ${(props) => props.theme.bg_page1};
	box-shadow: ${(props) => props.theme.boxShadow};
`;

export const Logo = styled(ReactComponent)`
	width: 50px;
	height: 50px;
	${flexCenter};
`;

export const Headline = styled.h1`
`;

export const Nav = styled.nav`
	flex: 1;
	${flexCenter};
`;

export const LoginContainer = styled.div`

`;