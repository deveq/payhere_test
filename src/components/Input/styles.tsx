import { Children } from 'react';
import styled from 'styled-components';
import { FiSearch } from 'react-icons/fi';
import { flexCenter } from 'styles/helper';

interface HistoryProps {
	show?: boolean;
}

export const Container = styled.form`
	${flexCenter};
	width: 250px;
	border-radius: 10px;
	margin-left: 20px;
	padding: 10px;
	border: 1px solid ${({ theme }) => theme.border3};
	background: ${({ theme }) => theme.bg_element1};
	position: relative;
`;

export const Input = styled.input`
	border: none;
	outline: none;
	flex: 1;
	font-size: 16px;
	color: ${({ theme }) => theme.text1};
	background: ${({ theme }) => theme.bg_element1};
`;

export const Icon = styled(FiSearch).attrs({
	'data-testid': 'icon'
})`
	width: 20px;
	height: 20px;
	color: ${({ theme }) => theme.text1};
	cursor: pointer;
	&:hover,
	&:active {
		opacity: 0.6;
	}
`;

export const History = styled.ul<HistoryProps>`
	display: ${({ show }) => (show ? 'flex' : 'none')};
	position: absolute;
	top: 45px;
	background: ${({ theme }) => theme.bg_element1};
	width: 100%;
	height: ${({ children }) => `${Children.count(children) * 40}`}px;
	border-radius: 10px;
	z-index: 10;
	box-shadow: ${({ theme }) => theme.boxShadow};
	padding: 10px 20px;
	box-sizing: border-box;
	flex-direction: column;
	justify-content: space-around;
`;

export const HistoryItem = styled.li`
	color: ${({ theme }) => theme.text1};
	font-size: 18px;
	display: flex;
	justify-content: space-between;
	max-width: 90%;
	white-space: nowrap;
	text-overflow: ellipsis;
	overflow: hidden;
	&:hover {
		opacity: 0.6;
	}
	cursor: pointer;
`;
