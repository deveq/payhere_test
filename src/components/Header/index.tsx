import * as Styled from './styles';
import BlindText from '../BlindText';
import Input from 'components/Input';
import { Link } from 'react-router-dom';
import NavMenu, {ILinkList} from '../NavMenu'
import Button from '../Button';

const navMenuList: ILinkList[] = [
    {title: 'issue', to: '/'},
    {title: 'repositories', to: '/repositories'},
    {title: 'settings', to: '/settings'},
]

const Header = () => {
	return (
		<Styled.Container>
			<Styled.Headline>
				<Link to="/">
					<Styled.Logo />
					<BlindText>페이히어</BlindText>
				</Link>
			</Styled.Headline>
			<Input placeholder="검색어를 입력해주세요" />
			<Styled.Nav>
                <NavMenu list={navMenuList} />
			</Styled.Nav>
			<Styled.LoginContainer>
                <Button>로그인</Button>
			</Styled.LoginContainer>
		</Styled.Container>
	);
};

export default Header;