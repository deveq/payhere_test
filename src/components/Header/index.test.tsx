import {screen, fireEvent} from '@testing-library/react';
import {renderWithWrapper} from 'tests';
import 'jest-styled-components'
import Header from './index';
import {lightTheme} from 'styles/theme';

describe('<Header />', () => {
    it('정상적으로 렌더링되는지 확인한다', () => {
        const {container} = renderWithWrapper(<Header />);

        const header = container.children[0];
        const headline = header.children[0];
        const input = header.children[1];
        const nav = header.children[2];

        expect(header).toBeInTheDocument();
        expect(headline).toBeInTheDocument();
        expect(input).toBeInTheDocument();
        expect(nav).toBeInTheDocument();

        const blindText = screen.getByText('페이히어');
        expect(blindText).toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });

    it('자식 컴포넌트들이 정상렌더링 되는지 확인 한다', () => {
        const {container} = renderWithWrapper(<Header />);

        const header = container.children[0];
        const headline = header.children[0];
        const input = header.children[1];
        const nav = header.children[2];

        expect(header).toBeInTheDocument();
        expect(headline).toBeInTheDocument();
        expect(input).toBeInTheDocument();
        expect(nav).toBeInTheDocument();

        const logo = screen.getByTestId('icon');
        expect(logo).toBeInTheDocument();

        const blindText = screen.getByText('페이히어');
        expect(blindText).toBeInTheDocument();

        expect(header).toHaveStyle({
            background: lightTheme.bg_page1,
            'box-shadow': lightTheme.boxShadow,
        });
    })

    it('이벤트이벤트', () => {
        const {container} = renderWithWrapper(<Header />);

        const header = container.children[0];
        const input = header.children[1];

        expect(input).toBeInTheDocument();
        
        fireEvent.change(input.children[0], { target: { value: 'test' } });
        expect(input.children[0]).toHaveValue('test');
        // fireEvent.submit(input);
        // TODO: recordHistory가 변경되었는지 조회하기
    })
})