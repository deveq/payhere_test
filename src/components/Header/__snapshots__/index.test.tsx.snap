// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> 정상적으로 렌더링되는지 확인한다 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0 35px;
  height: 70px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #F8F9FA;
  box-shadow: 0 5px 10px rgba(100,100,100,0.15),0 2px 4px rgba(100,100,100,0.15);
  position: relative;
  z-index: 10;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 50px;
  height: 50px;
}

.c1 path {
  opacity: 1;
}

.c1 path:nth-child(1) {
  fill: #A164F9;
}

.c1 path:nth-child(2) {
  opacity: 1;
  fill: #163BD8;
}

.c1 path:nth-child(3) {
  fill: #1D99FF;
}

.c7 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  position: absolute;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  width: 1px;
  height: 1px;
  margin: -1px;
  overflow: hidden;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 250px;
  border-radius: 10px;
  margin-left: 20px;
  padding: 10px;
  border: 1px solid #DEE2E6;
  background: #E9ECEF;
  position: relative;
}

.c4 {
  border: none;
  outline: none;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 16px;
  color: #212529;
  background: #E9ECEF;
}

.c5 {
  width: 20px;
  height: 20px;
  color: #212529;
  cursor: pointer;
}

.c5:hover,
.c5:active {
  opacity: 0.6;
}

.c6 {
  display: none;
  position: absolute;
  top: 45px;
  background: #E9ECEF;
  width: 100%;
  height: 0px;
  border-radius: 10px;
  z-index: 10;
  box-shadow: 0 5px 10px rgba(100,100,100,0.15),0 2px 4px rgba(100,100,100,0.15);
  padding: 10px 20px;
  box-sizing: border-box;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c9 {
  height: 30px;
  width: 100px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c10 {
  color: #212529;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 20px;
  text-transform: capitalize;
}

.c10:hover {
  opacity: 0.6;
}

.c10.active {
  font-weight: bold;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div>
  <header
    class="c0"
  >
    <h1
      class=""
    >
      <a
        href="/"
      >
        <svg
          class="c1"
        >
          logo.svg
        </svg>
        <span
          class="c2"
        >
          페이히어
        </span>
      </a>
    </h1>
    <form
      class="c3"
    >
      <input
        autocomplete="off"
        class="c4"
        placeholder="검색어를 입력해주세요"
        type="text"
        value=""
      />
      <svg
        class="c5"
        data-testid="icon"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="11"
          cy="11"
          r="8"
        />
        <line
          x1="21"
          x2="16.65"
          y1="21"
          y2="16.65"
        />
      </svg>
      <ul
        class="c6"
      />
    </form>
    <nav
      class="c7"
    >
      <ul
        class="c8"
      >
        <li
          class="c9"
        >
          <a
            aria-current="page"
            class="c10 active"
            href="/"
          >
            issue
          </a>
        </li>
        <li
          class="c9"
        >
          <a
            class="c10"
            href="/repositories"
          >
            repositories
          </a>
        </li>
        <li
          class="c9"
        >
          <a
            class="c10"
            href="/settings"
          >
            settings
          </a>
        </li>
      </ul>
    </nav>
  </header>
</div>
`;
